service/src/main/kotlin/fi/espoo/evaka/attendance/ChildAttendanceQueries.kt

```diff
@@ -75,22 +74,23 @@ fun Database.Transaction.insertAbsence(
         .first()
 }
 
-fun Database.Read.getChildAttendance(childId: UUID, unitId: DaycareId, date: LocalDate): ChildAttendance? {
+fun Database.Read.getChildAttendance(childId: UUID, unitId: DaycareId, now: HelsinkiDateTime): ChildAttendance? {
     // language=sql
     val sql =
         """
         SELECT id, child_id, unit_id, arrived, departed
         FROM child_attendance
-        WHERE child_id = :childId AND unit_id = :unitId AND tstzrange(:todayStart, :todayEnd, '[)') @> arrived
+        WHERE child_id = :childId AND unit_id = :unitId
+        AND (arrived::date = :date OR tstzrange(arrived, departed) @> :departedThreshold OR (arrived::date = departed::date AND arrived::date = :date))
```

Comment:
`arrived::date = :date` is listed in the first and third ORed condition, so the third condition seems to be unnecessary

Reply:
Yeah, you are right. I'm not sure what I was thinking anymore, so I can't say if the first condition is incomplete.

Reply:
But either way, `arrived::date = :date OR tstzrange(arrived, departed) @> :departedThreshold` should be fine here.